uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/BAO/LeaveRequest.php

```diff
@@ -534,22 +535,20 @@ private function deleteDates() {
    * @return array
    *   An array of formatted results
    */
-  public static function calculateBalanceChange($contactId, $fromDate, $fromType, $toDate = null, $toType = null) {
-
+  public static function calculateBalanceChange($contactId, DateTime $fromDate, $fromType, DateTime $toDate = null, $toType = null) {
     $leaveRequest = new self();
     $leaveRequest->contact_id = $contactId;
 
     //For single day leave requests
     if (!$toDate) {
-      $toDate = $fromDate;
+      $toDate = clone $fromDate;
     }
-    $startDate = new DateTime($fromDate);
-    $endDate = new DateTime($toDate);
-    $endDateUnmodified = new DateTime($toDate);
+
+    $endDateUnmodified = clone $toDate;
     // add one day to end date to include it in DatePeriod
-    $endDate->modify('+1 day');
+    $toDate->modify('+1 day');
     $interval   = new DateInterval('P1D');
-    $datePeriod = new DatePeriod($startDate, $interval, $endDate);
+    $datePeriod = new DatePeriod($fromDate, $interval, $toDate);
 
     $isHalfDay = ['half_day_am', 'half_day_pm'];
```

Comment:
Don't think so. They are names of reserved option values, so the risk of change is very low.